template/addons/env/env-auth.mjs

```diff
@@ -1,8 +1,24 @@
 import { z } from "zod";
 
-export const envSchema = z.object({
+export const serverEnvSchema = z.object({
   NEXTAUTH_SECRET: z.string(),
   NEXTAUTH_URL: z.string().url(),
   DISCORD_CLIENT_ID: z.string(),
   DISCORD_CLIENT_SECRET: z.string(),
 });
+
+export const clientEnvSchema = z.object({
+  // Specify your client-side environment variables schema here
+  // Be sure to name your environment variables with the prefix "NEXT_PUBLIC_"
+});
+
+/**
+ * Next.js client-side environment variables are evaluated at build time, 
+ * so only environment variables actually used will be included.
+ * 
+ * Define your client-side environment variables in this object to
+ * be able to use autocompletion and destructuring in your code. 
+ */
+export const clientEnv = {
```

Comment:
What does this do? Same for all installers

Reply:
Mentioned this earlier. This is a workaround for `process.env` not being a standard JS object client-side in Next.js. From the [docs](https://nextjs.org/docs/basic-features/environment-variables):
```
Note: In order to keep server-only secrets safe, environment variables are 
evaluated at build time, so only environment variables actually used will 
be included. This means that process.env is not a standard JavaScript object, 
so you’re not able to use object destructuring. Environment variables must be referenced as e.g. process.env.PUBLISHABLE_KEY, not const { PUBLISHABLE_KEY } = process.env.
```
By explicitly defining the variables in an actual js object you can get autocomplete when importing it in your code among other things and allows for the zod validation to work on the client.

Reply:
Hmm that sucks having to "duplicate" the keys but if its necessary it is. I tested some stuff out and this is as "minimal" I could get (compare last commit)...
https://github.com/juliusmarminge/trpc-rqv3